src/data/mv_polynomial.lean

```diff
@@ -195,6 +211,21 @@ finsupp.induction p
   (by have : M (C 0) := h_C 0; rwa [C_0] at this)
   (assume s a p hsp ha hp, h_add _ _ (this s a) hp)
 
+theorem induction_on' {M : mv_polynomial σ α → Prop} (p : mv_polynomial σ α) :
+    (∀ (u : σ →₀ ℕ) (a : α), M (monomial u a)) →
+    (∀ (p q : mv_polynomial σ α), M p → M q → M (p + q)) → M p :=
+begin
+  intros h1 h2,
+  apply finsupp.induction,
+    { have h1' : M (monomial 0 0) := h1 0 0,
+      rw monomial at h1', simp at h1', exact h1' },
+    intros a b f ha hb hMf,
+    apply h2,
+    {rw ←monomial, apply h1},
```

Comment:
`{apply h1}` works by itself (Lean does the unification). This proof is so close to a term mode proof apart from that one `simp`. Can it be made into a term mode proof? The advantage of a term mode proof is that it compiles more quickly.

Reply:
```lean
theorem induction_on' {P : mv_polynomial σ α → Prop} (p : mv_polynomial σ α)
    (h1 : ∀ (u : σ →₀ ℕ) (a : α), P (monomial u a))
    (h2 : ∀ (p q : mv_polynomial σ α), P p → P q → P (p + q)) : P p :=
finsupp.induction p (suffices P (monomial 0 0), by rwa monomial_zero at this,
                     show P (monomial 0 0), from h1 0 0) 
                    (λ a b f ha hb hPf, h2 _ _ (h1 _ _) hPf)
```

Can't seem to get rid of the `rwa`, but pretty much yes.